File: jawa/venator/src/main/java/ru/prohor/universe/venator/webhook/model/WebhookPayload.java
package ru.prohor.universe.venator.webhook.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;

@JsonIgnoreProperties(ignoreUnknown = true)
public record WebhookPayload(
        @NotNull
        String ref,
        @NotNull
        RepositoryInfo repository,
        @NotNull
        SenderInfo sender,
        @NotNull
        @JsonProperty("head_commit")
        CommitInfo headCommit
) {}

File: jawa/korriban/venator/src/main/kotlin/webhook/model/WebhookPayload.kt
package ru.prohor.universe.venator.webhook.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class WebhookPayload(
    val ref: String,
    val repository: RepositoryInfo,
    val sender: SenderInfo,
    @field:JsonProperty("head_commit")
    val headCommit: CommitInfo
)
